scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -325,70 +340,47 @@ def rmatvec(b):
 
     # Solve with preconditioner
     matvec_count = [0]
-    x, info = solver(A, b, M=precond, x0=x0, tol=tol)
+    if solver in [cg, cgs, bicg, bicgstab, gmres, qmr]:
+        x, info = solver(A, b, M=precond, x0=x0, rtol=tol)
+    else:
+        x, info = solver(A, b, M=precond, x0=x0, tol=tol)
```

Comment:
Same here as before: prefer to use `tol=` consistently with warning filter & todo